packages/squiggle-lang/src/rescript/Reducer/Reducer_Dispatch/Reducer_Dispatch_BuiltIn.res

```diff
@@ -1,72 +1,72 @@
-module CTV = Reducer_Extension.CodeTreeValue
-module Lib = Reducer_Extension.ReducerLibrary
-module ME = Reducer_MathJs.Eval
-module Rerr = Reducer_Error
+module ExternalLibrary = ReducerInterface.ExternalLibrary
+module MathJs = Reducer_MathJs
+open ReducerInterface.ExpressionValue
+open Reducer_ErrorValue
+
 /*
   MathJs provides default implementations for builtins
   This is where all the expected builtins like + = * / sin cos log ln etc are handled
   DO NOT try to add external function mapping here!
 */
-type codeTreeValue = CTV.codeTreeValue
-type reducerError = Rerr.reducerError
 
 exception TestRescriptException
 
-let callInternal = (call: CTV.functionCall): result<'b, reducerError> =>{
-
-  let callMatjJs = (call: CTV.functionCall): result<'b, reducerError> =>
+let callInternal = (call: functionCall): result<'b, errorValue> => {
+  let callMathJs = (call: functionCall): result<'b, errorValue> =>
     switch call {
-      | ("jsraise", [msg]) => Js.Exn.raiseError(CTV.show(msg)) // For Tests
-      | ("resraise", _) => raise(TestRescriptException) // For Tests
-      | call => call->CTV.showFunctionCall-> ME.eval
+    | ("jsraise", [msg]) => Js.Exn.raiseError(show(msg)) // For Tests
+    | ("resraise", _) => raise(TestRescriptException) // For Tests
+    | call => call->showFunctionCall->MathJs.Eval.eval
     }
 
   let constructRecord = arrayOfPairs => {
-      Belt.Array.map(arrayOfPairs, pairValue => {
+    Belt.Array.map(arrayOfPairs, pairValue => {
       switch pairValue {
-      | CTV.CtvArray([CTV.CtvString(key), valueValue]) =>
-        (key, valueValue)
-      | _ => ("wrong key type", pairValue->CTV.showWithType->CTV.CtvString)}
-      }) -> Js.Dict.fromArray -> CTV.CtvRecord -> Ok
+      | EvArray([EvString(key), valueValue]) => (key, valueValue)
+      | _ => ("wrong key type", pairValue->showWithType->EvString)
+      }
+    })
+    ->Js.Dict.fromArray
+    ->EvRecord
+    ->Ok
   }
 
-  let arrayAtIndex = (aValueArray: array<codeTreeValue>, fIndex: float) =>
+  let arrayAtIndex = (aValueArray: array<expressionValue>, fIndex: float) =>
```

Comment:
Because I am not supporting Integer values in the ReducerInterface.  To use integers we need support it there. There are 2 points to make this silliness.
a) Supporting integers and floats separately is not as simple as adding a new item to expressionValue variant. There is a whole can of worms there because of JavaScript. JavaScript does not have separate integers and floats. There are just numbers. So yes, JavaScript can deliver us a float as an index in fact.
b) I am not opening that can of worms yet because direct integer support is not necessary for the rest of the distribution library

Reply:
This issue can be properly handled on switching to our own parser.

Reply:
Remember. When we call JavaScript to execute number functions we really might get a float index.  Not doing the casting there moves the problem to somewhere else. Like having to insert cast operators automatically inside AST.

Reply:
RESOLVED